### DISCUSSION TITLE
Run isolated from filesystem

### DISCUSSION THREAD

**Discussion by adriangb** [duckdb/duckdb#11857] [Q&A]
By default DuckDB will access the filesystem or network to query things (e.g. `SELECT * FROM 'file.parquet'`). Using the Python SDK, there any way to create an isolated session that by default does not have access to anything and cannot be given access to anything via SQL? My goal is to run queries against a fixed dataset without letting those queries access other resources on the host system.

**Comment by adriangb** [comment]
I see there's the `disabled_filesystems` config option as well as `autoload_known_extensions` and `autoinstall_known_extensions`. I'm just not sure with what config I'd be "safe" from DuckDB accessing things I don't expect it to. I am sort of looking for the opposite of this: disable all access to everything and then only pass in a known dataset (in my case a pyarrow dataset) that should be the only thing queries can access.

**Reply by Tishj** [reply]
`enable_external_access` is something to look at

**Reply by adriangb** [reply]
That looks like what I’m looking for! Is there any extended documentation on what exactly the parameter does / how locked down it makes things?
